src/t_stream.c

```diff
@@ -2923,24 +2959,35 @@ void xpendingCommand(client *c) {
             return;
         }
 
-        if (startidx+3 < c->argc) {
-            /* 'consumer' was provided */
-            consumername = c->argv[startidx+3];
-        }
+	if (totalOption == 2) {
+	    consumername = c->argv[startidx+3];
+	    mkgroup = 1;
+	} else if(totalOption == 1) {
+	    if(!strcasecmp(c->argv[startidx+3]->ptr, "MKGROUP"))
+	        mkgroup = 1;
+	    else
+		consumername = c->argv[startidx+3];
+	}
     }
 
-    /* Lookup the key and the group inside the stream. */
-    robj *o = lookupKeyRead(c->db,c->argv[1]);
-    streamCG *group;
-
-    if (checkType(c,o,OBJ_STREAM)) return;
-    if (o == NULL ||
-        (group = streamLookupCG(o->ptr,groupname->ptr)) == NULL)
-    {
-        addReplyErrorFormat(c, "-NOGROUP No such key '%s' or consumer "
-                               "group '%s'",
-                               (char*)key->ptr,(char*)groupname->ptr);
-        return;
+    group = streamLookupCG(o->ptr,groupname->ptr);
+    if(group == NULL) {
+	if (!mkgroup) {
+	          addReplyErrorFormat(c, "-NOGROUP No consumer "
+                                       "group '%s'",(char*)groupname->ptr);
+                 return;
+	} 
+        streamID id;
+        id.ms = 0;
+        id.seq = 0;
+        group = streamCreateCG(o->ptr,groupname->ptr,strlen(groupname->ptr),&id,SCG_INVALID_ENTRIES_READ);
+        if(group) {
+            server.dirty++;
+            notifyKeyspaceEvent(NOTIFY_STREAM,"xgroup-create",key,c->db->id);
+        } else {
+            addReplyErrorFormat(c, "Create consumer group '%s' in XPENDING failed" , (char*)groupname->ptr);
+            return;
```

Comment:
again, error that can't actually happen. please drop (and fix indentation issues and tabs)